### PR TITLE
Fileio/stat `get_file_size` intrinsic in C be

### PR DESCRIPTION
**Pull request by WaelYoussfi**
For symlinks C `stat` resolves the link and returns the stats of the actual file/dir

### CODE REVIEW

File: src/dev/flang/be/c/Intrinsics.java [hunk 1]
```diff
@@ -123,7 +123,20 @@ public class Intrinsics extends ANY
...
-    put("fuzion.std.fileio.get_file_size", noFileIo); // NYI: #158
+    put("fuzion.std.fileio.get_file_size", (c,cl,outer,in) ->
+        {
+          var statIdent = new CIdent("statbuf");
+          var resultIdent = new CIdent("result");
+          return CStmnt.seq(
+            CExpr.decl("size_t", resultIdent, CExpr.int64const(-1)),
+            CExpr.decl("struct stat", statIdent),
+            // result = size if successful
+            CExpr.iff(CExpr.call("stat", new List<>(A0.castTo("char *"), statIdent.adrOf())).eq(CExpr.int8const(0)), resultIdent.assign(statIdent.field(new CIdent("st_size")))),
```
Comment (fridis) [inline src/dev/flang/be/c/Intrinsics.java:134]: This code suffers from #570. For now, can you add a NYI-comment referring to that issue, and then fix it for all issues.

**Review by fridis — COMMENTED** [review]
The code is basically fine, but I found an issue affecting all the handling of path names in the fileio code that should be referenced and fixed in another PR.

File: lib/fuzion/std/fileio.fz [hunk 2]
```diff
@@ -184,7 +184,7 @@ public fileio is
   public create_dir(
                 # the (relative or absolute) dir name, using platform specific path separators
                 path string) outcome unit is
-    arr := path.utf8.asArray
+    arr := path.utf8.asArray # NYI : #570
```
Comment (fridis) [inline lib/fuzion/std/fileio.fz:187]: please add a short comment if you reference an issue such that it is clear what the issue is (C string nul termination). 
#570 has been closed, so better update this patch to not introduce this problem again.

**Comment by fridis** [comment]
@WaelYoussfi, can you resolve the conflicts?

**Comment by WaelYoussfi** [comment]
> @WaelYoussfi, can you resolve the conflicts?

Resolved.

**Pull request merged**